pkg/service/grpc_service.go

```diff
@@ -25,10 +25,6 @@ type GRPCService struct {
 	gen.UnimplementedServiceServer
 }
 
-func (s *GRPCService) SetEval(eval eval.IEvaluator) {
-	s.eval = eval
-}
-
 func (s GRPCService) Serve(ctx context.Context, eval eval.IEvaluator) error {
 	s.eval = eval
```

Comment:
I never got time to finish reviewing before this got merged, why is the assignment being done this way?

Reply:
Assignment was previously being handled within the Serve method, for putting together tests access was needed to set a mocked eval. 
As per your comment I have moved the tests into the same package to allow access to the un-exported field, preventing the need for a setter, alternatively we can make eval available as an exported field and put the tests back in their own package.